macaddress/__init__.py

```diff
@@ -44,15 +44,3 @@ def format_mac(eui_obj, dialect):
             dialect = getattr(importlib.import_module(module), dialect_cls)
     eui_obj.dialect = dialect
     return str(eui_obj)
-
-
-from pkg_resources import get_distribution, DistributionNotFound
-import os.path
-
-try:
-    _dist = get_distribution('django-macaddress')
-except DistributionNotFound:
-    __version__ = 'Please install this project with setup.py'
-else:
-    __version__ = _dist.version
-VERSION = __version__   # synonym
```

Comment:
VERSION was added as per django-macaddress/django-macaddress/issues/7 (to provide version visible in Django--debug-toolbar)

removing it would be a regression, rather fix should be replacing the pkg_resource call with it's replacement

Reply:
From the pkg_resources docs:https://setuptools.pypa.io/en/latest/pkg_resources.html

>Use of pkg_resources is deprecated in favor of [importlib.resources](https://docs.python.org/3.11/library/importlib.resources.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata) and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Some useful APIs are also provided by [packaging](https://pypi.org/project/packaging) (e.g. requirements and version parsing). Users should refrain from new usage of pkg_resources and should work to port to importlib-based solutions.

Python 3.12 has removed pkg_resources from the standard library (moved to setuptools):
https://docs.python.org/3/whatsnew/3.12.html
> https://github.com/python/cpython/issues/95299: Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.

Reply:
here is one way of addressing this without additional libraries: https://github.com/googleapis/python-api-core/pull/80

Reply:
So, do you want i update the PR with this?